Use repl language tag for sample

abhro opened the pull request.

ablaom commented on docs/src/common_mlj_workflows.md:
````diff
@@ -14,9 +14,9 @@ channing = (Sex = rand(["Male","Female"], 462),
-import RDatasets
-channing = RDatasets.dataset("boot", "channing")
+```julia-repl
+julia> import RDatasets
+julia> channing = RDatasets.dataset("boot", "channing")
````
```suggestion
import RDatasets
channing = RDatasets.dataset("boot", "channing")
```

I think for consistency, we just skip all the `julia> ` prompts in common_mlj_workflows.jl`, no?

abhro replied:
Sounds good. It should be noted that when you copy paste into the REPL, as far as I've noticed (Julia 1.10) it ignores the prompt if it's from the clipboard. But for example workflows (without showing output) consistent lack of prompt also works.

ablaom commented on docs/src/learning_networks.md:
```diff
@@ -748,20 +748,19 @@ Q = @node sqrt(Z)
+julia> Xs = source(X)
+julia> rs = @node rows(Xs)
+julia> W = @node selectrows(Xs, @node shuffle(rs))
 
```
Generally, the pattern has been to only include  `julia> ` before input when corresponding output is to follow. Are you finding this confusing?

ablaom replied:
Good to have this feedback, thank you. 

Unfortunately, this convention is pretty widespread, at least in the MLJ ecosystem. I wonder if others are also finding this confusing. @OkonSamuel @EssamWisam What's you view of this? Do we need to include `julia>` prompts in every single line when there is sometimes REPL output to be included in a docstring? Often we just only add `julia>` before lines with output immediately following.

EssamWisam replied:
My preference is to minimize the use of `>julia` as it jams up the code (and has to be dropped before copy-pasting into `.jl` files) which as you said is consistent with the ongoing convention for `MLJ`. That said, I do find @abhro to be valid; it would be not immediately obvious to a beginner that the printed object is not an actual function being called and for that I think we could expose to readers the convention that what immediately comes after a line with `>julia` is the output of the code in that line.

In any case, in the future, we can explore the possibility of having the outputs be printed in a differently styled cell as in [Imbalance tutorials](https://juliaai.github.io/Imbalance.jl/dev/examples/smoten_mushroom/smoten_mushroom/) and [DataScienceTutorials](https://juliaai.github.io/DataScienceTutorials.jl/getting-started/model-tuning/) which would allow dropping all `>julia` and would be optimal for maximal facilitation of code reuse.

abhro replied:
Yes, I think the way the `@example` macro does it also provides a good separation between output and code.

ablaom commented on docs/src/mlj_cheatsheet.md:
````diff
@@ -96,20 +97,26 @@ y, X =  unpack(channing,
+```julia-repl
+julia> train, valid, test = partition(eachindex(y), 0.7, 0.2, rng=1234) # for 70:20:10 ratio
+```
 
````
I think in a cheatsheet we should just suppress all `julia>`. As far as I remember, all code is input (no output is displayed) for purposes of concision.

ablaom commented on docs/src/simple_user_defined_models.md:
```diff
@@ -51,32 +52,14 @@ MyRegressor(; lambda=0.1) = MyRegressor(lambda)
-    report=nothing
-    return fitresult, cache, report
-end
-MLJBase.predict(::MyRegressor, fitresult, Xnew) = MLJBase.matrix(Xnew) * fitresult
```
😆

ablaom commented on docs/src/transformers.md:
```diff
@@ -193,18 +193,17 @@ K-means clustering algorithm assigns one of three labels 1, 2, 3 to
-# transforming:
-Xsmall = transform(mach);
-selectrows(Xsmall, 1:4) |> pretty
+julia> # transforming:
```
This is really weird: a comment at the `julia>` prompt, here and later. Let's just skip all the prompts that don't precede a block of output (see previous comment).

abhro replied:
Yes, I'm not particularly happy with this change either, I just did it for consistency. Maybe there's a better way to write this?

abhro replied:
Maybe a better way to do this would be to split the code sample into 3, one for setup, one for transforming, and one for predicting?

ablaom replied:
Yes, you can re-organize so that the code comments become ordinary markdown text between seperate fenced blocks.

ablaom commented on docs/src/transformers.md:
```diff
@@ -231,7 +230,7 @@ compare[1:8]
 
-compare[51:58]
+julia> compare[51:58]
 8-element Array{Tuple{CategoricalValue{Int64,UInt32},CategoricalString{UInt32}},1}:
```
Yes! Here `julia>` is needed.

ablaom commented on docs/Project.toml:
```diff
@@ -15,6 +15,7 @@ MLJLinearModels = "6ee0df7b-362f-4a72-a706-9e79364fb692"
 MLJMultivariateStatsInterface = "1b6a4a23-ba22-4f51-9698-8599985d3728"
 Missings = "e1d29d7a-bbdc-5cf2-9ac0-f12de2c33e28"
 NearestNeighborModels = "636a865e-7cf4-491e-846c-de09b730eb36"
+ParallelKMeans = "42b8e9d4-006b-409a-8472-7f34b3fb58af"
```
What are we adding this?

ablaom commented:
@abhro My question regarding `ParallelMeans` notwithstanding, are you ready for a final review?

abhro commented:
Sure!

ablaom reviewed (approved):
Thanks for this work @abhro 🙏🏾

ablaom commented:
@EssamWisam Can you please go over the proposed changes to the cheatsheet.

@abhro Can you please the ParallelKMeans comment?

abhro commented:
Hello, I think I addressed it in the original comment thread? But to reiterate, this is to support the change in docs/src/transformers.md, where I changed the predicting transformers code samples to an `@example` block (lines 196–209), and it loads the `KMeans` model from the `ParallelKMeans` package.

EssamWisam reviewed (commented):
Reviewed the cheatsheet. All is okay. However, I wonder why only some of the backticks were replaced with \`\`\`julia \`\`\` and not others. For the purposes of viewing the cheatsheet in the website, it's better to use \`\`\`julia \`\`\` whenever the line is long enough.

Otherwise, the cheatsheet, viewed from the website, would look like this:
<img width="1314" alt="image" src="https://github.com/JuliaAI/MLJ.jl/assets/49572294/a7e0bdaf-02bd-4a5c-89c7-8586bcffbb6b">

where many lines are not syntax highlighted (e.g., look below Learning Curves).

Other great changes to facilitate viewing the cheatsheet from the browser:
- No overly long sections (chunk tuning into three ## headlines: ## Tuning model wrapper, ## Ranges for tuning (`range=...`), ## Tuning strategies)
- No overly long lines of code. For instance, the following is too long to view on the website. E.g., comments in:
<img width="846" alt="image" src="https://github.com/JuliaAI/MLJ.jl/assets/49572294/c2b07d04-b82e-4cd0-97ff-ca1650e94dbd">

@abhro let me know if you would be willing to revisit the cheat sheet accordingly. If not, I can consider that and add a commit.

For how the sheet views on the website after doing the change in the first bullet above:

[MLJ-Sheet.pdf](https://github.com/JuliaAI/MLJ.jl/files/15298946/MLJ-Sheet.pdf)



Thank you for the effort.

ablaom commented:
> ... Hello, I think I addressed it in the original comment thread? But to reiterate, this is to support the change in docs/src/transformers.md, where I changed the predicting transformers code samples to an @example block (lines 196–209), and it loads the KMeans model from the ParallelKMeans package

Thanks for the explanation. 

ParallelKMeans is not a regularly maintained package. Can we please change `pkg=ParallelKMeans` to `pkg=NearestNeighborModels`? That pkg is already in the Project.

abhro commented:
Hello, @EssamWisam, thank you for the feedback! I'm working on updating the cheatsheet to make it a little more consistent. Can I ask how you generated the pdf? It would help a lot in checking the changes I'm making. Thanks again!

abhro commented:
@ablaom I can't find the KMeans model in NearestNeighborsModel. I could use the one in Clustering/MLJClusteringInterface. Does that work?

ablaom commented:
> @ablaom I can't find the KMeans model in NearestNeighborsModel. I could use the one in Clustering/MLJClusteringInterface. Does that work?

Oops :flushed: Yes, use the MLJClusteringInterface version.

EssamWisam commented:
> Hello, @EssamWisam, thank you for the feedback! I'm working on updating the cheatsheet to make it a little more consistent. Can I ask how you generated the pdf? It would help a lot in checking the changes I'm making. Thanks again!

Well, the MLJ website is currently under development and (as of now only) unless you have some web development frameworks expertise, it's not straightforward to generate it.

That said, avoiding overly ong long lines, using \`\`\`julia \`\`\` more often and avoiding overlong long sections are pretty much all the conditions needed for the cheatsheet to render nicely.

abhro commented:
Alrighty! I think I've made those changes. If there's something I missed, or in some ways it could be made better, please don't hesitate to let me know! Thanks!

EssamWisam commented on docs/src/mlj_cheatsheet.md:
```diff
@@ -89,57 +93,75 @@ Same as above but exclude `:Time` column from `X`:
 y, X =  unpack(channing,
-               ==(:Exit),            # y is the :Exit column
-               !=(:Time);            # X is the rest, except :Time
+               ==(:Exit),     # y is the :Exit column
```
The comment makes it an overly long line for the cheat sheet. Should either exclucde the comment or write it after the code block.
```suggestion
               ==(:Exit),     
```

EssamWisam commented on docs/src/mlj_cheatsheet.md:
```diff
@@ -89,57 +93,75 @@ Same as above but exclude `:Time` column from `X`:
-               ==(:Exit),            # y is the :Exit column
-               !=(:Time);            # X is the rest, except :Time
+               ==(:Exit),     # y is the :Exit column
+               !=(:Time);     # X is the rest, except :Time
```
Same comment regarding the commend

EssamWisam commented on docs/src/mlj_cheatsheet.md:
````diff
@@ -191,13 +217,15 @@ or a list of pairs of row indices:
+```
 
-### Ranges for tuning (`range=...`)
+## Ranges for tuning (`range=...`)
````
So parenthesis aren't too large in the cheatsheet.
```suggestion
## Ranges for tuning `(range=...)`
```

EssamWisam commented on docs/src/mlj_cheatsheet.md:
```diff
@@ -65,11 +69,11 @@ type                                       | scitype
 Use `schema(X)` to get the column scitypes of a table `X`
 
-`coerce(y, Multiclass)` attempts coercion of all elements of `y` into scitype `Multiclass`
+- `coerce(y, Multiclass)` attempts coercion of all elements of `y` into scitype `Multiclass`
```
Should we mention, before this bullet, something like "Some example to coerce given data:"? Just thinking.

EssamWisam commented:
> Alrighty! I think I've made those changes. If there's something I missed, or in some ways it could be made better, please don't hesitate to let me know! Thanks!

Thank you for the valuable contribution, @abhro. I made some minor comments.
